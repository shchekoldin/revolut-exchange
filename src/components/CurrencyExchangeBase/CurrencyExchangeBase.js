// @flow
import React from 'react';
import cx from 'classnames/bind';
import money from 'revolut/lib/money';
import i18n from 'revolut/i18n';
import styles from './CurrencyExchangeBase.scss';

type Props = {
    baseCurrency: string,
    balance: number,
    baseAmount: number,
    onChange: (number) => void,
};

class CurrencyExchangeBase extends React.Component<Props> {
    render() {
        const {
            baseCurrency,
            balance,
            baseAmount,
            onChange,
        } = this.props;

        return (
            <div className={styles.CurrencyExchangeBase}>
                <div className={styles.currencyAmount}>
                    <div className={styles.label}>{baseCurrency}</div>
                    <input
                        className={cx(styles.control, styles.amount)}
                        type="text"
                        value={baseAmount}
                        maxLength={4}
                        pattern="[0-9]*"
                        autoFocus
                        onChange={(e) => {
                            const t = e.target;

                            onChange(t.validity.valid ? t.value : baseAmount);
                        }}
                    />
                </div>
                <div className={styles.sub}>
                    <div className={styles.currencyBalance}>
                        You have: {i18n.currency[baseCurrency]}{money.format(balance)}
                    </div>
                </div>
            </div>
        );
    }
}

export default CurrencyExchangeBase;
