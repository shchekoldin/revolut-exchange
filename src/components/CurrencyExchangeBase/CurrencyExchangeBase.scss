@import "~revolut/styles/_vars";

.CurrencyExchangeBase {
    flex: 1;
    padding: 2.5em 1.5em 0;
    box-sizing: border-box;
    min-width: 100%;

    .currencyAmount {
        overflow: hidden;

        .label {
            font-size: 40px;
            float: left;
            margin-top: 6px;
        }

        input.control {
            border: 1px solid lighten($body-color, 30%);
            background: transparent;
            width: 50%;
        }

        input.control:focus{
            outline: none;
        }

        .amount {
            font-size: 40px;
            text-align: right;
            float: right;
            color: $font-color;
        }
    }

    .sub {
        margin-top: 0.5rem;
        font-size: 0.9em;
        color: lighten($body-color, 20%);
        clear: both;
        font-weight: bold;
    }
}
