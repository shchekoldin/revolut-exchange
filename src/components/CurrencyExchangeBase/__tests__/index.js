import React from 'react';
import {shallow} from 'enzyme';
import chai from 'chai';
import CurrencyExchangeBase from '../CurrencyExchangeBase';

describe('Component <CurrencyExchangeBase /> should render without errors', () => {
    it('render with static props', () => {
        const noop = () => {
        };
        const props = {
            baseCurrency: 'EUR',
            balance: 100,
            baseAmount: 50,
            onChange: noop,
        };
        const wrapper = shallow(<CurrencyExchangeBase {...props} />);

        chai.expect(wrapper.instance()).to.be.not.undefined();
        chai.expect(wrapper.find('.currencyAmount').find('.label').text()).to.equal(props.baseCurrency);
        chai.expect(wrapper.find('.currencyAmount').find('.control.amount').prop('value')).to.equal(props.baseAmount);

        expect(wrapper).toMatchSnapshot();
    });
});
