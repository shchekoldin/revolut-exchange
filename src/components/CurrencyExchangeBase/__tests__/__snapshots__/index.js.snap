// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <CurrencyExchangeBase /> should render without errors render with static props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CurrencyExchangeBase
    balance={100}
    baseAmount={50}
    baseCurrency="EUR"
    onChange={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="currencyAmount"
        >
          <div
            className="label"
          >
            EUR
          </div>
          <input
            autoFocus={true}
            className="control amount"
            onChange={[Function]}
            step="0.01"
            type="number"
            value={50}
          />
        </div>,
        <div
          className="sub"
        >
          <div
            className="currencyBalance"
          >
            You have: 
            €
            100.00
          </div>
        </div>,
      ],
      "className": "CurrencyExchangeBase",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="label"
            >
              EUR
            </div>,
            <input
              autoFocus={true}
              className="control amount"
              onChange={[Function]}
              step="0.01"
              type="number"
              value={50}
            />,
          ],
          "className": "currencyAmount",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "EUR",
              "className": "label",
            },
            "ref": null,
            "rendered": "EUR",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "autoFocus": true,
              "className": "control amount",
              "onChange": [Function],
              "step": "0.01",
              "type": "number",
              "value": 50,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="currencyBalance"
          >
            You have: 
            €
            100.00
          </div>,
          "className": "sub",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "You have: ",
              "€",
              "100.00",
            ],
            "className": "currencyBalance",
          },
          "ref": null,
          "rendered": Array [
            "You have: ",
            "€",
            "100.00",
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="currencyAmount"
          >
            <div
              className="label"
            >
              EUR
            </div>
            <input
              autoFocus={true}
              className="control amount"
              onChange={[Function]}
              step="0.01"
              type="number"
              value={50}
            />
          </div>,
          <div
            className="sub"
          >
            <div
              className="currencyBalance"
            >
              You have: 
              €
              100.00
            </div>
          </div>,
        ],
        "className": "CurrencyExchangeBase",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="label"
              >
                EUR
              </div>,
              <input
                autoFocus={true}
                className="control amount"
                onChange={[Function]}
                step="0.01"
                type="number"
                value={50}
              />,
            ],
            "className": "currencyAmount",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "EUR",
                "className": "label",
              },
              "ref": null,
              "rendered": "EUR",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "autoFocus": true,
                "className": "control amount",
                "onChange": [Function],
                "step": "0.01",
                "type": "number",
                "value": 50,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="currencyBalance"
            >
              You have: 
              €
              100.00
            </div>,
            "className": "sub",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "You have: ",
                "€",
                "100.00",
              ],
              "className": "currencyBalance",
            },
            "ref": null,
            "rendered": Array [
              "You have: ",
              "€",
              "100.00",
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
